### PR TITLE
Implement signature verification

### PR DESCRIPTION
**Pull request by MarshalX**
2 tests are failing for now:
image

### CODE REVIEW

**Comment by MarshalX** [comment]
hi @DavidBuchanan314 @snarfed could you pls help me with fixing 2 tests? we should deny non-low-S signature somehow

**Comment by DavidBuchanan314** [comment]
Your `_encode_signature` method already extracts the `r` and `s` values, you just need to compare `s` against `N//2`, where N is the group modulus (I thiiiiiink, I'm slightly rusty...), and throw an exception or something if it doesn't. Note that N is different for both p256 and k256 curves

**Comment by DavidBuchanan314** [comment]
This is where I implement low-s *signing* in picopds, but I didn't do verification yet https://github.com/DavidBuchanan314/picopds/blob/main/signing.py

**Comment by DavidBuchanan314** [comment]
Also, for the record, I'm not 100% sure I got the check correct... it might be off-by-one, and I haven't written/found test cases to test the boundary cases.

**Comment by MarshalX** [comment]
@DavidBuchanan314 thank you! Now it passes tests. Could you pls explain where you get values of curve order?

**Comment by DavidBuchanan314** [comment]
Great question, you can find P-256 given in NIST SP 800-186 https://csrc.nist.gov/pubs/sp/800/186/final https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-186.pdf section 3.2.1.3.

SEC 2 specifies both P-256 and K-256 (aka secp256p1, secp256k1) https://www.secg.org/sec2-v2.pdf

File: packages/atproto_crypto/algs/base_alg.py [hunk 1]
```diff
@@ -34,11 +35,23 @@ def decompress_pubkey(self, pubkey: bytes) -> bytes:
             encoding=serialization.Encoding.X962, format=serialization.PublicFormat.UncompressedPoint
         )
 
-    @staticmethod
-    def _encode_signature(signature: bytes) -> bytes:
+    def _validate_dss_signature_s(self, s: int) -> None:
```
Comment (DavidBuchanan314) [inline packages/atproto_crypto/algs/base_alg.py:38]: Just as a code style thing, I'd recommend giving this function a more explicit name, e.g. `_ensure_dss_signature_low_s`. The word "validate" could mislead someone into thinking it does full signature verification, as opposed to just the low-s check.

Comment (MarshalX) [reply packages/atproto_crypto/algs/base_alg.py]: renamed!
